chore(Java) : add circular singly linked list

riteshyadav27 opened the pull request:
Singly Circular LinkedList with functionality of Add, Add in front, Display, Reverse , Search element in LinkedList

riteshyadav27 commented:
> Thanks for opening this pull request! Please check out our contributing guidelines.
> 

I checked that and all requirements were done.

riteshyadav27 commented:
I have added BFS and DFS code for graph traversal in java.

mohitchakraverty reviewed (changes requested):
Please also update the respective README.md file
In your case, it will be algorithms/Java/README.md

riteshyadav27 commented:
I updated the README.md file for circular singly linked list in Linked List Section , BFS and DFS in Graph Section .

mohitchakraverty reviewed (changes requested):
Please add more documentation to your code that helps users understand better.

riteshyadav27 commented:
I have added the Output of all programs with their proper input .

mohitchakraverty reviewed (changes requested):
- Please make different PRs for different algorithm
- In naming the file, replace underscore with dashes
- Change the name circularll to circular-singly-linkedlist
- Also add comments to your code explaining what or how your code is working

riteshyadav27 commented:
Updated Readme.md file, changed file name, explanation added for circular-singly-linkedlist having reverse(), addfront(), search() functions with Output

riteshyadav27 commented:
@mohitchakraverty will you please add hacktoberfest 2022 label 🙌

riteshyadav27 commented:
@mohitchakraverty I didn't understand why codespell is failed . Will you please help me what is it and how it works .

mohitchakraverty commented on algorithms/Java/linked-lists/circular-singly-linkedlist.java:
```diff
@@ -0,0 +1,119 @@
+
+*/
+
+// Here intial linkedlist was 9 5 8 7 1 2 and after using reverse function the linkedlist become 2 1 7 8 5 9
```
```suggestion
// Here initial linkedlist was 9 5 8 7 1 2 and after using reverse function the linkedlist become 2 1 7 8 5 9
```

mohitchakraverty reviewed (changes requested):
Also fix the spelling error in DSA/algorithms/CPlusPlus/Arrays/max-subarray-sum.cpp
At line line 23 : Approach

Then you are good to go and your code will pass all the test.

riteshyadav27 commented:
@mohitchakraverty Done. I updated both the files.

mohitchakraverty commented:
Great work! Thanks
